arrow-libs/optics/arrow-optics-compose/src/commonMain/kotlin/arrow/optics/Flow.kt

```diff
@@ -0,0 +1,70 @@
+package arrow.optics
+
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.FlowCollector
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.flow.StateFlow
+
+/**
+ * Exposes the values of [this] through the optic.
+ */
+public fun <T, A> SharedFlow<T>.optic(g: Getter<T, A>): SharedFlow<A> = object : SharedFlow<A> {
+  override suspend fun collect(collector: FlowCollector<A>): Nothing =
+    this@optic.collect { collector.emit(g.get(it)) }
+
+  override val replayCache: List<A>
+    get() = this@optic.replayCache.map { g.get(it) }
+}
+
+/**
+ * Exposes the values of [this] through the optic.
+ */
+public fun <T, A> StateFlow<T>.optic(g: Getter<T, A>): StateFlow<A> = object : StateFlow<A> {
+  override val value: A
+    get() = g.get(this@optic.value)
+
+  override suspend fun collect(collector: FlowCollector<A>): Nothing =
+    this@optic.collect { collector.emit(g.get(it)) }
+
+  override val replayCache: List<A>
+    get() = this@optic.replayCache.map { g.get(it) }
```

Comment:
Always contains single element.
```suggestion
    get() = listOf(value)
```

Reply:
I would prefer to keep this implementation, since it makes it more clear that we are just reusing the `replayCache` from the nested version.